pkg/ccl/changefeedccl/changefeed_dist.go

```diff
@@ -215,127 +234,179 @@ func startDistChangefeed(
 	if err != nil {
 		return err
 	}
+	cfKnobs := execCfg.DistSQLSrv.TestingKnobs.Changefeed
 
 	// Changefeed flows handle transactional consistency themselves.
 	var noTxn *kv.Txn
 
 	dsp := execCtx.DistSQLPlanner()
 	evalCtx := execCtx.ExtendedEvalContext()
-	planCtx := dsp.NewPlanningCtx(ctx, evalCtx, nil /* planner */, noTxn,
-		sql.DistributionTypeAlways)
-
-	var spanPartitions []sql.SpanPartition
-	if details.SinkURI == `` {
-		// Sinkless feeds get one ChangeAggregator on the gateway.
-		spanPartitions = []sql.SpanPartition{{SQLInstanceID: dsp.GatewayID(), Spans: trackedSpans}}
-	} else {
-		// All other feeds get a ChangeAggregator local on the leaseholder.
-		var err error
-		spanPartitions, err = dsp.PartitionSpans(ctx, planCtx, trackedSpans)
-		if err != nil {
-			return err
-		}
+
+	p, planCtx, err := makePlan(execCtx, jobID, details, initialHighWater, checkpoint, trackedSpans, selectClause)(ctx, dsp)
+	if err != nil {
+		return err
 	}
 
-	// Use the same checkpoint for all aggregators; each aggregator will only look at
-	// spans that are assigned to it.
-	// We could compute per-aggregator checkpoint, but that's probably an overkill.
-	aggregatorCheckpoint := execinfrapb.ChangeAggregatorSpec_Checkpoint{
-		Spans:     checkpoint.Spans,
-		Timestamp: checkpoint.Timestamp,
+	replanOracle := sql.ReplanOnChangedFraction(func() float64 { return replanChangefeedThreshold.Get(execCtx.ExecCfg().SV()) })
```

Comment:
nit: let's wrap things a bit.
Usually, go code should be under 100 characters or so.
You can move func() to another line and add returns after open brace